app/models/manageiq/providers/ibm_cloud/inventory/collector/power_virtual_servers/target_collection.rb

```diff
@@ -65,6 +65,18 @@ def sshkeys
     []
   end
 
+  def snapshots
+    @snapshots ||= references(:vms).map do |ems_ref|
+      snapshots_api.pcloud_cloudinstances_snapshots_getall(cloud_instance_id).snapshots.select do |snapshot|
+        snapshot.pvm_instance_id == ems_ref
+      end
```

Comment:
@jaywcarman it looks like this is getting a list of all snapshots for each vm target, if there are >1 targets this seems inefficient.  Could we get the list of snapshots first then loop through the `references(:vms)` ?  Maybe
```suggestion
    @snapshots ||= snapshots_api.pcloud_cloudinstances_snapshots_getall(cloud_instance_id).snapshots.select do |snapshot|
      references(:vms).include?(snapshot.pvm_instance_id)
    end
```